lib/config-base.d.ts

```diff
@@ -61,6 +61,12 @@ export interface HTTPOptions {
      * Used in node.js environments only.
      */
     connectTimeout?: number;
+    /**
+     * Custom DNS lookup function.
+     * Defaults to dns.lookup.
+     * Used in Node.js (>= v12.x) environment only.
```

Comment:
> Instead of making additional commits, the following commits can be reverted:
> * [`0894018` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/089401836f35c83e0fb157f18776a78bd758c127)
> * [`adb765c` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/adb765c0ac1f149c8ef8dc6902e914ff1c415e58)
> * [`276eb80` (#3524)](https://github.com/aws/aws-sdk-js/pull/3524/commits/276eb80c903d5bd41a60603d98adcc925cc72b9f)

In my suggestion, I'd given three commits to revert instead of suggesting at three places.
If incorporating suggestions is easy instead of reverting than let me post suggestions for others.

Reply:
```suggestion
```